src/mds/SimpleLock.cc

```diff
@@ -37,8 +37,8 @@ void SimpleLock::dump(ceph::Formatter *f) const {
   f->dump_int("num_wrlocks", get_num_wrlocks());
   f->dump_int("num_xlocks", get_num_xlocks());
   f->open_object_section("xlock_by");
-  if (get_xlock_by()) {
-    get_xlock_by()->dump(f);
+  if (auto mut = get_xlock_by(); mut) {
```

Comment:
The intrusive pointer `MutationRef()` should be parsed as `nullptr` already IMO.

Reply:
I think so too, a null MutationRef() should be boolean false?

Reply:
yeah.

Reply:
hmm.. there are two kinds of checks then - one which checks for nullptr and the other for `MutationRef()` instance. It's a bit confusing (esp. since its a shared_ptr).

Reply:
Sorry, you're confusing me Venky. Can you give an example of what you're concerned about?

Reply:
> Sorry, you're confusing me Venky. Can you give an example of what you're concerned about?

Ok. I haven't checked this closely, but in some places there is a mix of checks for the value returned by get_xlock_by() - nullptr vs MutationRef() and I wasn't 100% sure if the results of those checks are the same.